sklearn/tests/test_calibration.py

```diff
@@ -376,6 +377,31 @@ def decision_function(self, X):
     calibrated_clf.fit(X, y)
 
 
+def test_isotonic_calibration_strict():
+    """
+    Test that enforcing strict monotonicity for isotonic calibration
+    preserves rank-based metrics to handle issue
+    https://github.com/scikit-learn/scikit-learn/issues/16321
+    """
+
+    n_samples = 100
+    X, y = make_classification(n_samples=2 * n_samples, n_features=6,
+                               random_state=42)
+    X_train, y_train = X[:n_samples], y[:n_samples]
+    X_test, y_test = X[n_samples:], y[n_samples:]
```

Comment:
we can use `train_test_split` instead

Reply:
Fair enough, will do ! Then to uniformize things in the file, maybe the same should be applied to the other tests (I guess in the same uniformizing PR as for comments instead of docstrings).

Reply:
yes we could homogenize the file, in another PR